[Doc] Add benchmark guide (#635)
vllm-project/vllm-ascend: docs/source/developer_guide/evaluation/benchmark.md

> # Introduction

Reviewer: why two `#` level title?

Author: fixed

This document details the benchmark methodology for vllm-ascend, aimed at evaluating the performance under a variety of workloads. To maintain alignment with vLLM, we use the [benchmark](https://github.com/vllm-project/vllm/tree/main/benchmarks) script provided by the vllm project.

> # Run benchmarks

Reviewer: benchmark or benchmarks? which is the widely used one?

Author: fixed

> **Benchmark Coverage**: We measure offline e2e latency and throughput, and fixed-QPS online serving benchmarks, for more details see [vllm-ascend](https://github.com/vllm-project/vllm-ascend/tree/main/benchmarks).

Reviewer: for more details see [vllm-ascend benchmark scripts]

## 1. Run docker container
```bash
# Update DEVICE according to your device (/dev/davinci[0-7])
export DEVICE=/dev/davinci7

> export IMAGE=m.daocloud.io/quay.io/ascend/vllm-ascend:|vllm_ascend_version|

Reviewer: vllm_ascend_version doens't work here. Please follow other docs to make it correct.  i.e. you should add {code-block}

docker run --rm \
--name vllm-ascend \
--device $DEVICE \
--device /dev/davinci_manager \
--device /dev/devmm_svm \
--device /dev/hisi_hdc \
-v /usr/local/dcmi:/usr/local/dcmi \
-v /usr/local/bin/npu-smi:/usr/local/bin/npu-smi \
-v /usr/local/Ascend/driver/lib64/:/usr/local/Ascend/driver/lib64/ \
-v /usr/local/Ascend/driver/version.info:/usr/local/Ascend/driver/version.info \
-v /etc/ascend_install.info:/etc/ascend_install.info \
-v /root/.cache:/root/.cache \
-p 8000:8000 \
-e VLLM_USE_MODELSCOPE=True \
-e PYTORCH_NPU_ALLOC_CONF=max_split_size_mb:256 \
-it $IMAGE \
/bin/bash
```

## 2. Install dependencies
```bash
cd /workspace/vllm-ascend
pip config set global.index-url https://mirrors.tuna.tsinghua.edu.cn/pypi/web/simple
pip install -r benchmarks/requirements-bench.txt
```

## 3. (Optional)Prepare model weights
For faster running speed, we recommend downloading the model in advance：
```bash

> modelscope download --model LLM-Research/Meta-Llama-3.1-8B-Instruct

Reviewer: is the `modelscope` package installed in step 2? if not, you should make sure `modelscope` command is avaliable first.

Author: I'll add `modelscope` into [requirements](https://github.com/vllm-project/vllm-ascend/blob/main/benchmarks/requirements-bench.txt)

```

You can also replace all model paths in the [json](https://github.com/vllm-project/vllm-ascend/tree/main/benchmarks/tests) files with your local paths:
```bash
[
  {
    "test_name": "latency_llama8B_tp1",
    "parameters": {
      "model": "your local model path",
      "tensor_parallel_size": 1,
      "load_format": "dummy",
      "num_iters_warmup": 5,
      "num_iters": 15
    }
  }
]
```

## 4. Run benchmark script
Run benchmark script:
```bash

> bash benchmarks/scripts/run-performance-benchmarks.sh

Reviewer: make sure users are in vllm-ascend folder first.

Author: on the L32,  has located at `/workspace/vllm-ascend`

---------------Time to First Token----------------
Mean TTFT (ms):                          3394.37   
Median TTFT (ms):                        3359.93   
P99 TTFT (ms):                           3540.93   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          66.28     
Median TPOT (ms):                        64.19     
P99 TPOT (ms):                           97.66     
---------------Inter-token Latency----------------
Mean ITL (ms):                           56.62     
Median ITL (ms):                         55.69     
P99 ITL (ms):                            82.90     
==================================================

offline:
latency:
Avg latency: 4.944929537673791 seconds
10% percentile latency: 4.894104263186454 seconds
25% percentile latency: 4.909652255475521 seconds
50% percentile latency: 4.932477846741676 seconds
75% percentile latency: 4.9608619548380375 seconds
90% percentile latency: 5.035418218374252 seconds
99% percentile latency: 5.052476694583893 seconds

throughput:
Throughput: 4.64 requests/s, 2000.51 total tokens/s, 1010.54 output tokens/s
Total num prompt tokens:  42659
Total num output tokens:  43545
```

> For more details, detailed json file is generated to the path `benchmark/results`

Reviewer: ```suggestion
The result json files are generated into the path `benchmark/results`
 ```

Author: done

# Introduction

> This document outlines the benchmarking process for vllm-ascend, designed to evaluate its performance under various workloads. To make consistency with the vllm community, we have reused the vllm community [benchmark](https://github.com/vllm-project/vllm/tree/main/benchmarks) script.

Author: grammar is ok know
